### PR TITLE
PWX-37598: failed to resize cloud drive to: 246 due to: [mh500-3-ds02-b] fcd/_00ae/7d5f9de0847046459ca83a791d272b0e.vmdk due to: File []/vmfs/volumes/6658caf2-5c39344a-ab46-3cecef4f0aee/fcd/_00ae/7d5f9de0847046459ca83a791d272b0e.vmdk was not found

### PR DESCRIPTION
**Pull request by yjoshi-px**
**What this PR does / why we need it**:
Modified the pool expansion error message and added the svmotion warning message in the error message whenever we get this type of error where vmdk is missing

**Which issue(s) this PR fixes** (optional)
Closes # PWX-37598

**Testing**
Screenshot 2024-06-28 at 8 31 23 AM

### CODE REVIEW

File: vsphere/vsphere.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ const (
 	vSphereDataStoreLock = "vsphere-ds-lock"
 	configProperty       = "config.hardware"
 	permissionError      = "Permission to perform this operation was denied"
+	svmotionErrorMsg     = "Retry pool expansion if you have performed svmotion on the disk"
```
Comment (nrevanna) [inline vsphere/vsphere.go:35]: "retry pool expansion, if a storage vMotion operation was in progress during expansion"

File: vsphere/vsphere.go [hunk 2]
```diff
@@ -629,12 +630,19 @@ func (ops *vsphereOps) Expand(
 
 	task, err := vm.Reconfigure(ctx, spec)
 	if err != nil {
+		if strings.Contains(err.Error(), ".vmdk not found") {
```
Comment (nrevanna) [inline vsphere/vsphere.go:633]: ".vmdk not found" make this a constant variable

File: vsphere/vsphere.go (hunk 1 above)
Comment (vinayakshnd) [reply vsphere/vsphere.go]: Done

File: vsphere/vsphere.go (hunk 2 above)
Comment (vinayakshnd) [reply vsphere/vsphere.go]: Done

File: vsphere/vsphere.go [hunk 3]
```diff
@@ -629,12 +631,19 @@ func (ops *vsphereOps) Expand(
 
 	task, err := vm.Reconfigure(ctx, spec)
 	if err != nil {
+		if strings.Contains(err.Error(), vmdkNotFoundErrorMsg) {
```
Comment (nrevanna) [inline vsphere/vsphere.go:634]: - Let's do a logrus.Errorf(err) before this if statement

File: vsphere/vsphere.go [hunk 4]
```diff
@@ -629,12 +631,19 @@ func (ops *vsphereOps) Expand(
...
+			return 0, fmt.Errorf(err.Error() + svmotionErrorMsg)
+		}
 		return 0, err
 	}
 
 	err = task.Wait(ctx)
 	if err != nil {
-		return 0, fmt.Errorf("error resizing vmdk: %s due to:  %s", vmdkPath, err)
+		errMsg := fmt.Errorf("error resizing vmdk: %s due to:  %s", vmdkPath, err)
```
Comment (nrevanna) [inline vsphere/vsphere.go:642]: - print this errMsg here
- errMsg = fmt.Errorf("error resizing vmdk: %s. Path not found. Retry pool expansion, if a storage vMotion operation was in progress during expansion.", vmdkPath)
- Declare the error string as a global variable and use in line 635 as well.

**Review by nrevanna — COMMENTED** [review]
Thanks for addressing the previous review comments @vinayakshnd . Took a second look and feel we could improve the message a bit more.

**Review by nrevanna — APPROVED** [review]

**Pull request merged**